_blogposts/2023-09-15-uncurried-mode.mdx

```diff
@@ -0,0 +1,127 @@
+---
+author: rescript-team
+date: "2023-09-15"
+title: Uncurried Mode
+badge: roadmap
+description: |
+  A tour of new capabilities coming to ReScript v11
+---
+
+> This is the fourth post covering new capabilities that'll ship in ReScript v11. You can check out the first post on [Better Interop with Customizable Variants](/blog/improving-interop), the second post on [Enhanced Ergonomics for Record Types](/blog/enhanced-ergonomics-for-record-types) and the third post on [First-class Dynamic Import Support](/blog/first-class-dynamic-import-support).
+
+## Introduction
+
+ReScript is a language that strives to keep its users free from experiencing runtime errors. Usually, when a program compiles, it will already do what the user described.
+But there is still a concept in the language that makes it easy to let some errors slip through. Currying! 
+
+Because of currying, partial application of functions is possible. That feature is always advertised as something really powerful. For instance, 
+
+```rescript
+let add = (a, b) => a + b
+let addFive = add(5)
+```
+
+is shorter than having to write all remaining parameters again 
+
+```rescript
+let add = (a, b) => a + b
+let addFive = (b) => add(5, b)
+```
+
+Also, currying is what enabled ReScript to have optional (`~param=?`) or default (`~param=true`) parameters. 
```

Comment:
Maybe remove this sentence? It might be confusing, as we can still have optional and default parameters even in uncurried mode.